Hudl.Mjolnir/Config/MjolnirConfiguration.cs

```diff
@@ -100,11 +102,11 @@ public CommandConfiguration GetCommandConfiguration(string groupKey = null)
         /// <param name="key">Bulkhead configuration for a given key. Default value returned if non-existent or 
         /// null.</param>
         /// <returns></returns>
-        public BulkheadConfiguration GetBulkheadConfiguration(string key = null)
+        public BulkheadConfiguration GetBulkheadConfiguration(GroupKey key)
         {
-            if (key == null) return DefaultBulkheadConfiguration;
+            if (string.IsNullOrEmpty(key.Name)) return DefaultBulkheadConfiguration;
             BulkheadConfiguration bulkheadConfiguration;
-            return BulkheadConfigurations.TryGetValue(key, out bulkheadConfiguration) ?
+            return BulkheadConfigurations.TryGetValue(key.Name, out bulkheadConfiguration) ?
```

Comment:
Is it possible for the Dictionary to have a null value for any key? It might be good to protect against returning `null` values from this method to prevent `NullReferenceExceptions` when callers access properties on the returned value, since the calling code assumes non-null objects returned.